Library/Homebrew/formula_installer.rb

```diff
@@ -517,9 +517,11 @@ def check_conflicts
   # Compute and collect the dependencies needed by the formula currently
   # being installed.
   def compute_dependencies
+    return @compute_dependencies if @compute_dependencies
+
     req_map, req_deps = expand_requirements
     check_requirements(req_map)
-    expand_dependencies(req_deps + formula.deps)
+    @compute_dependencies = expand_dependencies(req_deps + formula.deps)
```

Comment:
I'd probably do something like
```ruby
@compute_dependencies ||= begin
  req_map, req_deps = expand_requirements
  check_requirements(req_map)
  expand_dependencies(req_deps + formula.deps)
end
```

but don't have strong feelings if you do otherwise?